### PR TITLE
Provide Kubeflow XGBoostJob support for MultiKueue

### PR DESCRIPTION
**Pull request by mszadkow**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
The PR introduces a new MultiKueue adapter to handle XGBoostJob (Kubeflow).
We want to extend MultiKueue capabilities to satisfy the needs of early adopters.

#### Which issue(s) this PR fixes:

Relates #2552

#### Does this PR introduce a user-facing change?

```release-note
MultiKueue: Support for the Kubeflow XGBoostJob
```

### CODE REVIEW

File: pkg/controller/jobs/kubeflow/jobs/xgboostjob/xgboostjob_controller.go [hunk 1]
```diff
@@ -45,14 +45,15 @@ func init() {
...
 	}))
 }
 
 // +kubebuilder:rbac:groups=scheduling.k8s.io,resources=priorityclasses,verbs=list;get;watch
 // +kubebuilder:rbac:groups="",resources=events,verbs=create;watch;update;patch
 // +kubebuilder:rbac:groups=kubeflow.org,resources=xgboostjobs,verbs=get;list;watch;update;patch
-// +kubebuilder:rbac:groups=kubeflow.org,resources=xgboostjobs/status,verbs=get;update
-// +kubebuilder:rbac:groups=kubeflow.org,resources=xgboostjobs/finalizers,verbs=get;update
+// +kubebuilder:rbac:groups=kubeflow.org,resources=xgboostjobs/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=kubeflow.org,resources=xgboostjobs/finalizers,verbs=get;update;patch
```
Comment (trasc) [inline pkg/controller/jobs/kubeflow/jobs/xgboostjob/xgboostjob_controller.go:56]: why do we need this?

File: pkg/controller/jobs/kubeflow/jobs/xgboostjob/xgboostjob_multikueue_adapter.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+	kueuealpha "sigs.k8s.io/kueue/apis/kueue/v1alpha1"
+	"sigs.k8s.io/kueue/pkg/controller/constants"
+	"sigs.k8s.io/kueue/pkg/controller/jobframework"
+	"sigs.k8s.io/kueue/pkg/util/api"
+	clientutil "sigs.k8s.io/kueue/pkg/util/client"
+)
+
+type multikueueAdapter struct{}
+
+var _ jobframework.MultiKueueAdapter = (*multikueueAdapter)(nil)
```
Comment (mimowo) [inline pkg/controller/jobs/kubeflow/jobs/xgboostjob/xgboostjob_multikueue_adapter.go:40]: The adapter code is almost identical between the integrations. I would like to put effort at some point to commonize it, at least for Kubeflow. @mszadkow could you attempt commonizing them in this PR, or in a follow up?

Comment (mszadkow) [reply pkg/controller/jobs/kubeflow/jobs/xgboostjob/xgboostjob_multikueue_adapter.go:40]: Sure I will make the attempt, I would prefer in follow up

Comment (mimowo) [reply pkg/controller/jobs/kubeflow/jobs/xgboostjob/xgboostjob_multikueue_adapter.go:40]: Sg, I will prefer to commonize the adapters for the 2 integrations we currently have before adding new ones.

Comment (mszadkow) [reply pkg/controller/jobs/kubeflow/jobs/xgboostjob/xgboostjob_multikueue_adapter.go:40]: Ok, I will do that thing first

Comment (mimowo) [reply pkg/controller/jobs/kubeflow/jobs/xgboostjob/xgboostjob_multikueue_adapter.go:40]: to be clear it is ok in a follow up to this pr.

**Comment by mimowo** [comment]
/lgtm
/approve
Thanks!

File: pkg/controller/jobs/kubeflow/jobs/xgboostjob/xgboostjob_multikueue_adapter.go (hunk 2 above)
Comment (tenzen-y) [reply pkg/controller/jobs/kubeflow/jobs/xgboostjob/xgboostjob_multikueue_adapter.go:40]: +1
Commonizing the adapters would be better.

**Pull request merged**